modules/ROOT/pages/patterns/reference.adoc

```diff
@@ -788,536 +846,911 @@ The following table shows variants of the quantifier syntax and their canonical
+
+The pattern in the path selector function must be a variable-length relationship and not a quantified path pattern.
+
+.Not allowed
+[source]
 ----
-(:X&Y)
+shortestPath(((a)-[:R]-(b)){1,5})
+
+shortestPath((:A)-->+(:B))
 ----
-[source, role=noheader]
+
+[[shortest-functions-rules-path-pattern-length]]
+==== Path pattern length
+
+There must be exactly one relationship pattern in the path pattern.
+
+.Allowed
+[source]
 ----
-(:X&A)-[:R]->(:B&Y)
+shortestPath((a)-[:R*1..5]-(b))
 ----
 
-[[variable-length-relationships]]
-== Variable-length relationships
+.Not allowed
+[source]
+----
+shortestPath((a)-[:R*1..5]-(b)-->(:X))
 
-Prior to the introduction of the syntax for quantified path patterns and quantified relationships in Neo4j 5.9, the only way in Cypher to match paths of variable length was with a variable-length relationship.
-This syntax is still available.
-It is equivalent to the syntax for quantified relationships, with the following differences:
+shortestPath((:A))
 
-* Position and syntax of quantifier.
-* Semantics of the asterisk symbol.
-* Type expressions are limited to the xref::syntax/operators.adoc#query-operators-boolean[disjunction operator].
-* The xref:clauses/where.adoc[WHERE] clause is not allowed.
+allShortestPaths((a:A)-[:S*]->(:B), (a)-[:R*1..3]->(:C))
+----
 
-[[variable-length-relationships-syntax]]
-=== Syntax
+[[shortest-functions-rules-filtering]]
+==== Pre and post filtering
 
-[source, syntax, role=noheader]
+If the `MATCH` clause of the `shortestPath()` function includes a `WHERE` clause, this condition will act as a pre-filter: paths satisfying the `WHERE` clause are first found, and from those paths the shortest path is selected.
+
+.Examples of pre-filters:
+[source]
 ----
-varLengthRelationship ::=
-    "<-[" varLengthFiller "]-" 
-  | "-[" varLengthFiller "]->"
-  | "-[" varLengthFiller "]-"
-varLengthFiller ::= [ relationshipVariable ] [ varLengthTypeExpression ] 
-    [ varLengthQuantifier ] [ propertyKeyValueExpression ]
+MATCH p = shortestPath(()-[*]-())
+WHERE all(n in nodes(p) WHERE n.p < 42)
 
-varLengthTypeExpression ::= ":" varLengthTypeTerm
+MATCH p = shortestPath(()-[*]-())
+WHERE all(r in relationships(p) WHERE r.p < 42)
+----
 
-varLengthTypeTerm ::=
-    typeIdentifier
-  | varLengthTypeTerm "|" varLengthTypeTerm
+These contrast with `WHERE` clauses that are not part of the same `MATCH` clause.
 
-varLengthQuantifier ::= varLengthVariable | varLengthFixed
+.Example of post-filters
+[source]
+----
+MATCH p = shortestPath(()-[*]-())
+WITH nodes(p) AS N
+WHERE all(n in N WHERE n.p < 42)
+----
 
-varLengthVariable ::= "*" [ [ lowerBound ] ".." [ upperBound ] ]
+[[shortest-functions-rules-non-determinism]]
+==== Non-determinism
 
-varLengthFixed ::= "*" fixedBound
+If there is more than one path with minimum length, then `shortestPath()` function will return one of those paths non-deterministically.
+As `allShortestPaths()` would return all of those paths, its results are deterministic.
+
```

Comment:
There is also a rule about `shortestPath()` not allowing to find shortest cycles, that is a shortest path from a node to itself.